pkg/storagepolicy/sdkstoragepolicy.go

```diff
@@ -0,0 +1,263 @@
+		return nil, status.Error(codes.InvalidArgument, "Must supply a Storage Policy Name")
+	}
+
+	_, err := p.kv.Delete(prefixWithName(req.GetName()))
+	if err != kvdb.ErrNotFound && err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to delete Storage Policy %s: %v", req.GetName(), err)
+	}
+
+	return &api.SdkOpenStoragePolicyDeleteResponse{}, nil
+}
+
+func (p *SdkPolicyManager) Inspect(
+	ctx context.Context,
+	req *api.SdkOpenStoragePolicyInspectRequest,
+) (*api.SdkOpenStoragePolicyInspectResponse, error) {
+	if req.GetName() == "" {
+		return nil, status.Error(codes.InvalidArgument, "Must supply a Storage Policy Name")
+	}
+
+	var volSpecs *api.VolumeSpecUpdate
+	kvp, err := p.kv.GetVal(prefixWithName(req.GetName()), &volSpecs)
+	if err == kvdb.ErrNotFound {
+		return nil, status.Errorf(codes.NotFound, "Policy %s not found", req.GetName())
+	} else if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to get policy %s information: %v", req.GetName(), err)
+	}
+
+	err = jsonpb.Unmarshal(strings.NewReader(string(kvp.Value)), volSpecs)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Json Unmarshal failed for policy %s: %v", req.GetName(), err)
+	}
+
+	return &api.SdkOpenStoragePolicyInspectResponse{
+		StoragePolicy: &api.SdkStoragePolicy{
+			Name:   req.GetName(),
+			Policy: volSpecs,
+		},
+	}, nil
+}
+
+func (p *SdkPolicyManager) Enumerate(
+	ctx context.Context,
+	req *api.SdkOpenStoragePolicyEnumerateRequest,
+) (*api.SdkOpenStoragePolicyEnumerateResponse, error) {
+
+	// get all keyValue pair at /storage/policy/policies
+	kvp, err := p.kv.Enumerate(policyPrefix + policyPath)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to get policies from database: %v", err)
+	}
+
+	policies := make([]*api.SdkStoragePolicy, 0)
+	for _, policy := range kvp {
+		volSpecs := &api.VolumeSpecUpdate{}
+		err = jsonpb.Unmarshal(strings.NewReader(string(policy.Value)), volSpecs)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Json Unmarshal failed for policy %s: %v", policy.Key, err)
+		}
+		storagePolicy := &api.SdkStoragePolicy{
+			Name:   strings.TrimPrefix(policy.Key, policyPrefix+policyPath+"/"),
+			Policy: volSpecs,
+		}
+		policies = append(policies, storagePolicy)
+	}
+
+	return &api.SdkOpenStoragePolicyEnumerateResponse{
+		StoragePolicies: policies,
+	}, nil
+}
+
+func (p *SdkPolicyManager) Enforce(
+	ctx context.Context,
+	req *api.SdkOpenStoragePolicyEnforceRequest,
+) (*api.SdkOpenStoragePolicyEnforceResponse, error) {
+	if req.GetName() == "" {
+		return nil, status.Error(codes.InvalidArgument, "Must supply a Storage Policy Name")
+	}
+
+	policy, err := p.Inspect(ctx,
+		&api.SdkOpenStoragePolicyInspectRequest{
+			Name: req.GetName(),
+		},
+	)
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "Policy with name %s not found", req.GetName())
+	}
+
+	m := jsonpb.Marshaler{}
+	policyStr, err := m.MarshalToString(policy.StoragePolicy)
```

Comment:
We should not place the spec in Enforcement since this is a copy. If the original is then changed, this version still has the old information.

Instead this should only save the name of the policy as a pointer to the actual policy managed by this system. That way when it is updated, then GetEnforce..() is called, it gets the latest changes.